spec/watch_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'filewatch/watch'
+require 'stud/temporary'
+
+Thread.abort_on_exception = true
+
+describe FileWatch::Watch do
+  let(:directory) { Stud::Temporary.directory }
+  let(:file_path) { File.join(directory, "1.log") }
+  let(:loggr)     { double("loggr", :debug? => true) }
+  let(:results)   { [] }
+  let(:quit_proc) do
+    lambda do
+      Thread.new do
+        sleep 1
+        subject.quit
+      end
+    end
+  end
+  let(:subscribe_proc) do
+    lambda do
+      subject.subscribe(0.1, 4) do |event, path|
+        results.push([event, path])
+      end
+    end
+  end
+  let(:log_simulation_proc) do
```

Comment:
I'd remove the simulation from the proc name give it something more spec-related like write_two_lines